modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/src/main/java/com/liferay/headless/batch/engine/internal/resource/v1_0/ExportTaskResourceImpl.java

```diff
@@ -185,6 +186,9 @@ private List<String> _toList(String fieldNamesString) {
 	private BatchEngineExportTaskLocalService
 		_batchEngineExportTaskLocalService;
```

Comment:
I think we should use the service in the rest of the existing cases in this resource impl (get by ERC, post, etc), as otherwise, we would be allowing any user to get the Export Task by its external reference code or requesting a new export without having permission over the company

Reply:
Hi @ccorreagg ,
As discussed, this fix will contain the protection against cross-company references, while the role-based permissions will be discussed later.
Also added the integration tests for the new methods.